src/group_theory/congruence.lean

```diff
@@ -0,0 +1,414 @@
+/-
+Copyright (c) 2019 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import group_theory.submonoid data.setoid algebra.pi_instances data.equiv.algebra
+
+/-!
+# Congruence relations
+
+This file defines multiplicative and additive congruence relations, first by extending
```

Comment:
Could you say here what you mean by a congruence relation? (It is not a standard term to me, and I can imagine several natural interpretations).